Author: Piotr Engelking <inkerman42@gmail.com>
Description: Disable the filtering packets with incorrect checksum.
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=372536
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/urlsnarf.c
+++ b/urlsnarf.c
@@ -245,6 +245,7 @@
 	extern char *optarg;
 	extern int optind;
 	int c;
+	struct nids_chksum_ctl chksum_ctl;
 	
 	while ((c = getopt(argc, argv, "i:p:nvh?V")) != -1) {
 		switch (c) {
@@ -305,6 +306,12 @@
                 }
         }
 
+        chksum_ctl.netaddr = 0;
+        chksum_ctl.mask = 0;
+        chksum_ctl.action = NIDS_DONT_CHKSUM;
+
+        nids_register_chksum_ctl(&chksum_ctl, 1);
+
 	nids_run();
 	
 	/* NOTREACHED */
